src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -147,6 +148,8 @@ void IslandWindow::_HandleCreateWindow(const WPARAM, const LPARAM lParam) noexce
     ShowWindow(_window.get(), nCmdShow);
 
     UpdateWindow(_window.get());
+
+    UpdateWindowIconForActiveMetrics(_window.get());

```

Comment:
did you also mean to call this from WM_THEMECHANGED ?

Reply:
I did, but I chose to defer that until we also get WM_SETTINGSCHANGED.